File: 03-Refrescos/src/main/java/es/joseluisgs/dam/model/Maquina.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package es.joseluisgs.dam.model;

import java.io.Serializable;
import java.util.ArrayList;

/**
 *
 * @author link
 */
public class Maquina implements Serializable{
     private ArrayList<Refresco> listaRefrescos;
    
    public Maquina(){
        listaRefrescos = new ArrayList<Refresco>();
    }
    
    public synchronized void add(Refresco r){
        listaRefrescos.add(r);
    }
    
    public synchronized Refresco get(int i){
        return listaRefrescos.get(i);
    }
    
    public synchronized Refresco remove(int i){
        return listaRefrescos.remove(i);
    }
    
    public synchronized Maquina removeAll(){
        Maquina lista = new Maquina();
        while(listaRefrescos.size() > 0){
            lista.add(listaRefrescos.remove(0));
        }
        return lista;
    }
    
    public synchronized int size(){
        return listaRefrescos.size();
    }
}

File: 11-Refrescos-Kotlin/src/main/kotlin/model/Maquina.kt
package es.joseluisgs.dam.clienteservidor.model

import java.io.Serializable


class Maquina : Serializable {
    private val listaRefrescos: ArrayList<Refresco> = ArrayList<Refresco>()

    // Lo mejor es hacerlo con candados Lock de hilos!!
    @Synchronized
    fun add(r: Refresco) {
        listaRefrescos.add(r)
    }

    @Synchronized
    operator fun get(i: Int): Refresco {
        return listaRefrescos[i]
    }

    @Synchronized
    fun remove(i: Int): Refresco {
        return listaRefrescos.removeAt(i)
    }

    @Synchronized
    fun removeAll(): Maquina {
        val lista = Maquina()
        while (listaRefrescos.size > 0) {
            lista.add(listaRefrescos.removeAt(0))
        }
        return lista
    }

    @Synchronized
    fun size(): Int {
        return listaRefrescos.size
    }

}
